src/DateTimePickerBase/DateTimePickerBase.js

```diff
@@ -306,15 +310,23 @@ module.exports = kind(
 	/**
 	* @private
 	*/
-	refresh: function (){
+	refresh: function () {
 		this.destroyClientControls();
 		this.pickers = null;
 		delete this._tf;
 		if (this.value) {
-			this.localeValue = dateFactory({unixtime: this.value.getTime(), timezone: "local"});
+			this.localeValue = dateFactory({unixtime: this.value.getTime(), timezone: 'local'});
 		}
 		this.set('open', false);
 		this.initDefaults();
 		this.render();
+	},
+
+	/**
+	* @private
+	*/
+	backKeyHandler: function () {
+		if (this.open) this.set('value', this._initialValue);
```

Comment:
There is a long-standing guard against firing the change event for a falsy value via this commit: https://github.com/enyojs/moonstone/commit/a721bdec0cc124bae8b5e2cf72e3437e5d05b432. It seems like this guard is no longer necessary as we are no longer handling the `onChange` event ourselves, so I've removed it to address your concern. Let me know what you think!